src/func.d

```diff
@@ -1004,9 +1004,18 @@ extern (C++) class FuncDeclaration : Declaration
                     if (fdv.isFinalFunc())
                         error("cannot override final function %s", fdv.toPrettyChars());
 
-                    doesoverride = true;
                     if (!isOverride())
-                        .error(loc, "cannot implicitly override base class method %s with %s; add 'override' attribute", fdv.toPrettyChars(), toPrettyChars());
+                    {
+                        if (fdv.isFuture())
+                        {
+                            .deprecation(loc, "@future base class method %s is being overridden by %s; rename the latter", fdv.toPrettyChars(), toPrettyChars());
+                            // Treat 'this' as an introducing function, giving it a separate hierarchy in the vtbl[]
```

Comment:
Generally, adding a new virtual to a base class is only done if there's an immediate major problem, and my proposal allows this to get it done and the problem solved.

The special case is there in any case - it's hard to have a function in the symbol table and yet selectively ignore it.